File: mobile/src/types/adminReport.ts
export interface AdminReportSummary {
  todaysTreatments: number;
  completedTreatments: number;
  cancelledTreatments: number;
  patientsVisited: number;
  totalClaims: number;
  pendingClaims: number;
  approvedClaims: number;
  rejectedClaims: number;
  totalKm: number;
  totalTravelAmount: number;
  averageKmPerTherapist: number;
  activeTherapists: number;
  topPerformingTherapist: string | null;
  generatedAt: string;
  periodLabel: string;
  trendPeriodLabel: string;
  hasData: boolean;
  trends: ReportTrendPoint[];
  claimsByStatus: ReportClaimStatusPoint[];
  topTherapists: ReportTopTherapist[];
  recentActivity: ReportActivity[];
  insights: ReportInsight[];
}

export type ReportClaimStatus =
  | "all"
  | "pending"
  | "approved"
  | "rejected";

export type ReportActivityType =
  | "claim"
  | "treatment"
  | "travel";

export type ReportInsightDirection = "up" | "down" | "neutral";

export interface ReportTrendPoint {
  date: string;
  completedTreatments: number;
  totalKm: number;
  travelAmount: number;
}

export interface ReportClaimStatusPoint {
  status: Exclude<ReportClaimStatus, "all">;
  count: number;
}

export interface ReportTopTherapist {
  therapistId: number;
  therapistName: string;
  completedTreatments: number;
  totalKm: number;
  claimsSubmitted: number;
}

export interface ReportActivity {
  id: string;
  activityType: ReportActivityType;
  therapistName: string;
  occurredAt: string;
  status: string;
  amount: number | null;
  description: string;
}

export interface ReportInsight {
  key: string;
  title: string;
  value: string;
  detail: string;
  direction: ReportInsightDirection;
  changePercent: number | null;
}

export interface AdminReportFilters {
  fromDate: string | null;
  status: ReportClaimStatus;
  therapistId: number | null;
  therapistName: string | null;
  toDate: string | null;
}

File: backend/app/schemas/admin_report.py
from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel


class ReportKpis(BaseModel):
    todays_treatments: int
    completed_treatments: int
    cancelled_treatments: int
    patients_visited: int
    total_claims: int
    pending_claims: int
    approved_claims: int
    rejected_claims: int
    total_km: float
    total_travel_amount: float
    average_km_per_therapist: float
    active_therapists: int
    top_performing_therapist: str | None


class ReportTrendPoint(BaseModel):
    date: date
    completed_treatments: int
    total_km: float
    travel_amount: float


class ReportClaimStatusPoint(BaseModel):
    status: Literal["pending", "approved", "rejected"]
    count: int


class ReportTopTherapist(BaseModel):
    therapist_id: int
    therapist_name: str
    completed_treatments: int
    total_km: float
    claims_submitted: int


class ReportActivity(BaseModel):
    id: str
    activity_type: Literal["claim", "treatment", "travel"]
    therapist_name: str
    occurred_at: datetime
    status: str
    amount: float | None
    description: str


class ReportInsight(BaseModel):
    key: str
    title: str
    value: str
    detail: str
    direction: Literal["up", "down", "neutral"]
    change_percent: float | None = None


class AdminReportOverview(BaseModel):
    generated_at: datetime
    period_label: str
    trend_period_label: str
    has_data: bool
    kpis: ReportKpis
    trends: list[ReportTrendPoint]
    claims_by_status: list[ReportClaimStatusPoint]
    top_therapists: list[ReportTopTherapist]
    recent_activity: list[ReportActivity]
    insights: list[ReportInsight]
